front-end/src/components/Post/Post.tsx

```diff
@@ -182,6 +183,7 @@ const Post = ( { className, data, isMotion = false, isProposal = false, isRefere
 					onchainLink={definedOnchainLink}
 					status={postStatus}
 				/>
+				{onchainId ? <Poll postId={post.id} /> : null}
```

Comment:
I don't understand why we check for `onChainId` here. Also we should use `{condtion && <Component>}` if you want to display something/nothing conditionnaly.

Reply:
We want to show polls on only onchain posts right? Thus onChainId check if its an on chain post. Changed to use &&.

Reply:
Added a comment as well

Reply:
> We want to show polls on only onchain posts right? Thus onChainId check if its an on chain post. Changed to use &&.

No, it's only relevant for discussions, **not** on-chain discussions. Also copying what I wrote previously:
 >Not any discussion should have a poll. Only users that want a poll on their discussion should have one.
> This would be a function in the create post form, with a toggle "Add a poll to this discussion".

Reply:
done